src/analyze.rs

```diff
@@ -153,13 +173,15 @@ fn extract_dominated_subgraph(
         (dominated, not_dominated)
     };
 
-    // Cheap sanity checks
-    assert!(reachable.len() <= graph.node_count());
-    assert!(dominator_addrs.len() <= graph.node_count());
-    assert!(dominator_addrs.len() <= dominators.len());
-    assert!(dominator_addrs.len() <= reachable.len());
-    assert_eq!(dominated.node_count() + rest.len(), reachable.len());
-    assert!(dominated.node_count() <= dominator_addrs.len() + 1);
+    if reachable.len() > graph.node_count()
```

Comment:
I wonder if some of these checks should be moved up, that is return early if error.

Reply:
i think you're right that many of them could be, except for the last two, "dominated", as that is computed on the statement before this.

We can do return early in a future PR